pkg/system/dependencies.go

```diff
@@ -17,4 +19,18 @@ type SystemDependencies struct {
 	Proxy        *traefik.Traefik
 	Scope        string
 	StartTime    string
+	instancesMap sync.Map
```

Comment:
What is the reason we use a `Map` here as we don't use the value of the `Map`?

Reply:
it is for the quick search in its keys. bc if we use a slice, then we have to iterate over all of it to get the answer, but a Map uses a hash-table which is very quick in finding the keys.
Here we also use a `sync.Map` which is multi thread safe compare to the standard go map.